src/Transform/Monomorphize.hs

```diff
@@ -162,8 +163,9 @@ makeInstantiations xis (SProgFun x [] [] tp tm) =
     [ProgFun x [] (renameCalls xis tm) (renameCallsTp xis tp)]
 makeInstantiations xis (SProgFun x tgs ys tp tm) =
   let tiss = Map.toList (xis Map.! x) in
-    map (\ (tis, i) ->
-           let s = Map.fromList (zip (tgs ++ ys) (SubTp <$> tis)) in
+    map (\ ((tags, tis), i) ->
```

Comment:
The name `tags` is chosen awkwardly, revealing that we in principle want 4 names: 2 for the tag/type distinction * 2 for the argument/parameter distinction.

Reply:
I wondered about that...elsewhere I've used `tgs` and `tgs'` because for tags, arguments and parameters seem almost symmetric.